src/qibolab/pulses.py

```diff
@@ -1400,7 +1402,60 @@ def modulated_waveforms(self):  #  -> tuple[Waveform, Waveform]:
 
     @property
     def serial(self):
-        return f"FluxPulse({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
+        if isinstance(self.qubit, Coupler):
+            return f"Coupler_FluxPulse({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
+        else:
+            return f"FluxPulse({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
+
+
+class CouplerFluxPulse(Pulse):
```

Comment:
I tried but I ran into something like this I think https://stackoverflow.com/a/16310777. It crashed because the `init` from the `FluxPulse` got `type=PulseType.COUPLERFLUX` before reaching the `Pulse`.

Reply:
I see, then one solution is to hardcode the type as a class variable
```py
class FluxPulse(Pulse):
    PULSE_TYPE = PulseType.FLUX

    def __init__(self, ...):
        super().__init__(..., type=self.PULSE_TYPE, ...)


class CouplerFluxPulse(FluxPulse):
    PULSE_TYPE = PulseType.COUPLERFLUX
    # no need to overwrite __init__
```

Otherwise, you can always call a the `Pulse.__init__` directly instead of using `super()`,
```py
class CouplerFluxPulse(FluxPulse):
    def __init__(...):
        Pulse.__init__(...)
```

I think the first is simpler if it works.